server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{number::next_number, requisition::common::get_lines_for_requisition};
+
+use super::{RequisitionTransferProcessor, RequisitionTransferProcessorRecord};
+use chrono::Utc;
+use repository::{
+    NumberRowType, RepositoryError, Requisition, RequisitionLineRow, RequisitionLineRowRepository,
+    RequisitionRow, RequisitionRowRepository, RequisitionRowStatus, RequisitionRowType,
+    StorageConnection,
+};
+use util::uuid::uuid;
+
+pub struct CreateResponseRequisitionProcessor;
+impl RequisitionTransferProcessor for CreateResponseRequisitionProcessor {
+    fn get_description(&self) -> String {
+        "Create response requisition from request requisition".to_string()
+    }
+
+    fn try_process_record(
+        &self,
+        connection: &StorageConnection,
+        record_for_processing: &RequisitionTransferProcessorRecord,
+    ) -> Result<Option<String>, RepositoryError> {
+        // Check can execute
+        let RequisitionTransferProcessorRecord {
+            linked_requisition,
+            requisition: source_requisition,
+            ..
+        } = &record_for_processing;
+
+        if source_requisition.requisition_row.r#type != RequisitionRowType::Request {
+            return Ok(None);
+        }
+
+        if source_requisition.requisition_row.status != RequisitionRowStatus::Sent {
+            return Ok(None);
+        }
+
+        if linked_requisition.is_some() {
+            return Ok(None);
+        }
+
```

Comment:
Great idea, my plan originally was to outsource this to diagrams, but have to have comments in code to at the very least justify why this would only run once: cdf1c83ffd1f795961007396268e33155aa798e8

Reply:
fix comment: b136e629844022b71e5c5bf7b5e1fb18051a7543
fix typo: d4eccc768eeb74fda962d890fea0617056933d26

Reply:
Also did the same for shipments, and noticed that it is possible to have a double update on shipment (although it wouldn't cause infinite loop but still not ideal, fixed with new condition in `UpdateInboundShipmentProcessor`: 59567f157f4b08f12057fbe7eae923f6a9f58b1f, basically Outbound Shipment is linked, and updated, which causes Inbound shipment to be updated, condition was also restricted to update InboundShipment only if OutboundShipment status was changed).

I would also like to remove a little bit confusion by calling shipment/requisitions with better names within processors (don't use source and linked, just go with request and response, inbound and outbound), I'll try tomorrow

Reply:
Ohh man, i ended up updating requisition and shipments with more specific variable naming, and did a couple of extras
* Use destructure + `.map` for requisition line generation
* Fix comments
* Use destructure for requisition update (saw it in one of @roxy-dao PRs, cleaner syntax then `mut` and slightly safer with immutability i think)

Anyways the commit is here: 61227a0f9fc0d16f041a48a5bf7202316a7a9390, this makes the total diff from the last time you looked at this pr this: https://github.com/openmsupply/open-msupply/compare/5743790cbce106555035d67741b1528fe4f2eb11...61227a0f9fc0d16f041a48a5bf7202316a7a9390

Reply:
Looks awesome @andreievg!  Those inline logic comments are super helpful.